workflow/torch/requires_grad.py

```diff
@@ -2,19 +2,28 @@
 
 
 @contextmanager
-def requires_nograd(module):
+def requires_grad(module, grad=True):
```

Comment:
I would assert that grad is bool, could be kinda tricky bug to hunt down

Reply:
I'll add an assert when the other branch is merged into this

Reply:
Added check on grad